packages/solc-transpiler/src/compiler.ts

```diff
@@ -463,3 +488,157 @@ const transpileContract = (
+  ) => {
+    const placeholderStart: number = linkLocation.start * 2 // 2x because this is expressed in bytes but we will operate on hex string
+    const placeholderLength: number = linkLocation.length * 2 // 2x because this is expressed in bytes but we will operate on hex string
+    const placeholderEnd: number = placeholderLength + placeholderStart
+    const prevBytecodeString: string = bytecodeObject.object
+    const originalPlaceholderString: string = prevBytecodeString.slice(
+      placeholderStart,
+      placeholderEnd
+    )
+    log.debug(
+      `Parsed ${placeholderIndex}th link for file ${fileName}, AKA library ${libraryName} with placeholder ${originalPlaceholderString} at elements (${placeholderStart},${placeholderEnd}) bytecode string.`
+    )
+    bytecodeObject.object =
+      prevBytecodeString.slice(0, placeholderStart) +
+      getPlaceholderHexString(libraryName) +
+      prevBytecodeString.slice(placeholderEnd)
+    originalPlaceholderStrings.set(libraryName, originalPlaceholderString)
+    placeholderIndex++
+  }
+
+  executeOnAllLinks(
+    solcOutput.evm.bytecode,
+    pushOriginalPlaceholderAndSubstituteValidHex,
+    'REPLACE BYTECODE INVALID HEX STRINGS WITH VALID'
+  )
+  executeOnAllLinks(
+    solcOutput.evm.deployedBytecode,
+    pushOriginalPlaceholderAndSubstituteValidHex,
+    'REPLACE DEPLOYED BYTECODE INVALID HEX STRINGS WITH VALID'
+  )
+
+  return originalPlaceholderStrings
+}
+
+/**
+ * Executes the given callback on each linkReference (library metadata) for a given contractJSONOutput.bytecode or .deployedBytecode object
+ *
+ * @param bytecodeObjectFromSolcOutput The given contractJSONOutput.bytecode or .deployedBytecode object
+ * @param callback The callback to execute
+ * @param callbackDescription Optional logging string describing the functionality this callback will perform
+ */
+const executeOnAllLinks = (
+  bytecodeObjectFromSolcOutput: any,
+  callback: (
+    bytecodeObject: any,
+    linkLocation: any,
+    fileName: string,
+    libraryName: string
+  ) => void,
+  callbackDescription: string = '(UNSPECIFIED)'
+) => {
+  log.debug(
+    `asked to execute callback performing functionality: [${callbackDescription}] on all JSON links...`
+  )
+  const linkRefs = bytecodeObjectFromSolcOutput.linkReferences
+  for (const fileName in linkRefs) {
+    // tslint:disable
+    const libraryName: string = Object.keys(linkRefs[fileName])[0]
+    log.debug(`Parsing links for ${libraryName}...`)
+    for (const linkLocation of linkRefs[fileName][libraryName]) {
+      callback(
+        bytecodeObjectFromSolcOutput,
+        linkLocation,
+        fileName,
+        libraryName
+      )
+    }
+  }
+}
+
+/**
+ * Gets a deterministic, collision-resistant valid hex string placeholder for a given input string
+ *
+ * @param libraryName The string for which to get a deterministic hex string.
+ * @param byteLength Number of bytes worth of hex string to return.
+ */
+const getPlaceholderHexString = (
```

Comment:
This function is not general purpose, so I'd just name it how it's being used so it is very clear what it does: `getTranspiledLibraryPlaceholder(libraryName: string)`

Reply:
Don't worry about this -- will update later.